usecases/traverser/target_vector_param_helper.go

```diff
@@ -19,13 +19,7 @@ import (
 	"github.com/weaviate/weaviate/entities/schema"
 )
 
-type TargetVectorParamHelper struct{}
-
-func NewTargetParamHelper() *TargetVectorParamHelper {
-	return &TargetVectorParamHelper{}
-}
-
-func (t *TargetVectorParamHelper) GetTargetVectorOrDefault(sch schema.Schema, className, targetVector string) (string, error) {
+func GetTargetVectorOrDefault(sch schema.Schema, className, targetVector string) (string, error) {
```

Comment:
I would revert this change and use `TargetVectorParamHelper` struct, this way the code is encapsulated in one struct which purpose is to properly parse target vector settings. I'm against decoupling this to "hanging" methods as then when time goes we tend to forget what those do. I'm a fan of building blocks and encapsulation. which can be easily passed from one service to another

Reply:
The code isn't encapsulated, it's another file in the traverser namespace.  This is the opposite of encapsulated, it is increasing the already large number of files we have in there.  I will be happy to move it to a helper package.

> I'm against decoupling this to "hanging" methods as then when time goes we tend to forget what those do.

I cannot understand this at all.

> I'm a fan of building blocks and encapsulation. which can be easily passed from one service to another

So am I.  There is no encapsulated data here, which is already a red flag.  What are you encapsulating?

The go standard practice for building blocks and encapsulation is to put the encapsulated code in a package, and then import it.  This provides more namespace protection than mixing a helper class into an important class.  Mixing classes together is the opposite of encapsulation.

In addition, you are reducing code safety and security by adding a nil pointer to a critical data structure.  We have all agreed that we need to improve code quality and program stability, so I would like to know why you are insisting on keeping unsafe code that can panic and potentially corrupt data.